wasm2c/examples/fac/fac.c

```diff
@@ -317,35 +315,38 @@ static u32 w2c_fac(u32 w2c_p0) {
     w2c_i1 = w2c_p0;
     w2c_i2 = 1u;
     w2c_i1 -= w2c_i2;
-    w2c_i1 = w2c_fac(w2c_i1);
+    w2c_i1 = w2c_fac(module_instance, w2c_i1);
     w2c_i0 *= w2c_i1;
   }
   FUNC_EPILOGUE;
   return w2c_i0;
 }
 
-static void init_memory(void) {
+static void init_globals(Z_fac_module_instance_t* module_instance) {
+}
+static void init_memory(Z_fac_module_instance_t* module_instance) {
 }
 
-static void init_table(void) {
+static void init_table(Z_fac_module_instance_t* module_instance) {
   uint32_t offset;
 }
 
 /* export: 'fac' */
-u32 (*Z_facZ_fac)(u32);
-
-static void init_exports(void) {
-  /* export: 'fac' */
-  Z_facZ_fac = (&w2c_fac);
+u32 Z_facZ_fac(Z_fac_module_instance_t* module_instance, u32 w2c_p0) {
+  return w2c_fac(module_instance, w2c_p0);
 }
 
-void Z_fac_init(void) {
+void Z_fac_init_module(void) {
+  assert(wasm_rt_is_initialized());
   init_func_types();
-  init_globals();
-  init_memory();
-  init_table();
-  init_exports();
 }
 
-void Z_fac_free(void) {
+void Z_fac_instantiate(Z_fac_module_instance_t* module_instance) {
+  assert(wasm_rt_is_initialized());
```

Comment:
Could we also assert that `Z_fac_init_module` has been called?

Reply:
Done (by adding a static bool here: https://github.com/fixpointOS/wabt/blob/module_instancing/src/template/wasm2c.declarations.c#L317)